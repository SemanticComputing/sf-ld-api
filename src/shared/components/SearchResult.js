import React from 'react';
import _ from 'lodash';
import $ from 'jquery';
import ReactTooltip from 'react-tooltip';
import eli from '../lib/eli';
import { string } from 'prop-types';

export default class SearchResult extends React.Component {

  constructor(props) {
    const showContentTypes = [
      'http://data.finlex.fi/schema/sfl/Section',
      'http://data.finlex.fi/schema/sfl/Subsection'
    ];
    super(props);
    this.sectionId = eli.getSectionOfALawLocalId(props.workUrl);
    this.statuteId = eli.getStatuteLocalId(props.workUrl);
    this.statuteTitle = props.statuteTitle || '';
    this.heading = props.title;
    this.content = (_.indexOf(showContentTypes, props.type) > -1) ? this.sanitize(props.content, props.query) : '';
    this.versionUrl = props.versionUrl;
    this.workUrl = props.workUrl;
    this.statuteVersionUrl = props.statuteVersionUrl;
  }

  sanitize(html, query) {
    const $html = $('<div />', {html:html});
    $html.find('.item-identifier').html('');
    $html.find('.reference-amendment').html('');
    $html.find('.heading').html('');
    $html.find('.section-id').html('');
    $html.html($html.html().replace(new RegExp(query,'gi'), '<strong>$&</strong>'));
    return $html.html();
  }

  render() {
    return (
      <div className="search-result">
        <div className="search-result-statute-id"><a href={this.statuteVersionUrl ? this.statuteVersionUrl : this.versionUrl}>{this.statuteId}</a></div>
        <div className="search-result-statute-title"><a href={this.statuteVersionUrl ? this.statuteVersionUrl : this.versionUrl}>{this.statuteTitle}</a></div>
        <div className="search-result-title"><a href={this.versionUrl}>{this.sectionId} {this.heading && this.sectionId ? '- '+this.heading : this.heading ? this.heading : ''}</a></div>
        <div className="search-result-text" dangerouslySetInnerHTML={{__html: this.content}}></div>
        <ReactTooltip />
        <hr/>
      </div>
    );
  }
}

SearchResult.propTypes = {
  workUrl: string,
  content: string,
  versionUrl: string,
  statuteVersionUrl: string,
  type: string,
  query: string,
  title: string,
  statuteTitle: string
};
