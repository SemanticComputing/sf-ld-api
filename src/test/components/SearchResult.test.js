import React from 'react';
import { shallow, mount, render } from 'enzyme';

import SearchResult from '../../shared/components/SearchResult';

const props = {
  'title': 'Ammattitoiminnassa tehty rikos',
  'content': '<div class=\"section\"><h4 class=\"item-identifier\">25 §</h4><span class=\"reference reference-amendment\"><a href=\"http://data.finlex.fi/eli/sd/2006/301/alkup\">(28.4.2006/301)</a></span>\n<h4 class=\"heading\">Ammattitoiminnassa tehty rikos</h4>\n<div class=\"subsection\"><p class=\"text-element\">Jos eläinlääkärinammatin harjoittaja on tuomittu vankeusrangaistukseen rikoksesta, jonka hän on tehnyt ammattitoiminnassaan, ja tuomioistuimen tuomio on saanut lainvoiman ja jos rikokseen liittyvistä asianhaaroista on havaittavissa, ettei hän ole sen luottamuksen arvoinen, jota hänen tulee nauttia, Elintarviketurvallisuusvirasto voi poistaa eläinlääkärinammatin harjoittajalta määräajaksi tai, jos asianhaarat ovat erittäin raskauttavat, lopullisesti ammatinharjoittamisoikeuden.</p>\n</div>\n<div class=\"subsection\"><p class=\"text-element\">Jos valtion tai kunnan taikka kuntayhtymän virassa oleva eläinlääkärinammatin harjoittaja on rikoksen johdosta tuomittu erotettavaksi virantoimituksesta tai viralta pantavaksi, noudatetaan vastaavasti, mitä 1 momentissa säädetään.</p>\n</div>\n<div class=\"subsection\"><p class=\"text-element\">Tuomioistuimen tulee viipymättä lähettää jäljennös 1 ja 2 momentissa tarkoitettua asiaa koskevasta pöytäkirjasta ja tuomiosta Elintarviketurvallisuusvirastolle.</p>\n</div>\n<div class=\"subsection\"><p class=\"text-element\">Jo ennen kuin tuomioistuimen tuomio, jolla eläinlääkärinammatin harjoittaja on tuomittu vankeusrangaistukseen taikka viralta pantavaksi tai virantoimituksesta erotettavaksi, on saanut lainvoiman, Elintarviketurvallisuusvirasto voi kieltää eläinlääkärinammatin harjoittajaa harjoittamasta ammattia.</p>\n</div>\n</div>',
  'query': 'rikos',
  'workUrl': 'http://data.finlex.fi/eli/sd/2000/29/luku/5/pykala/25',
  'versionUrl': 'http://data.finlex.fi/eli/sd/2000/29/luku/5/pykala/25/ajantasa/20060501',
  'statuteVersionUrl': 'http://data.finlex.fi/eli/sd/2000/29/ajantasa/20160101',
  'statuteTitle': 'Laki eläinlääkärinammatin harjoittamisesta',
  'type': 'http://data.finlex.fi/schema/sfl/Section'
};

describe('<SearchResult />', () => {
  it('renders the given result', () => {
    const wrapper = shallow(<SearchResult {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
