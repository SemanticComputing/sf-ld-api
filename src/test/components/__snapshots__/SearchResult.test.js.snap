// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchResult /> renders the given result 1`] = `
<div
  className="search-result"
>
  <div
    className="search-result-statute-id"
  >
    <a
      href="http://data.finlex.fi/eli/sd/2000/29/ajantasa/20160101"
    >
      29/2000
    </a>
  </div>
  <div
    className="search-result-statute-title"
  >
    <a
      href="http://data.finlex.fi/eli/sd/2000/29/ajantasa/20160101"
    >
      Laki eläinlääkärinammatin harjoittamisesta
    </a>
  </div>
  <div
    className="search-result-title"
  >
    <a
      href="http://data.finlex.fi/eli/sd/2000/29/luku/5/pykala/25/ajantasa/20060501"
    >
      5 luku 25 § 
       
      - Ammattitoiminnassa tehty rikos
    </a>
  </div>
  <div
    className="search-result-text"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<div class=\\"section\\"><h4 class=\\"item-identifier\\"></h4><span class=\\"reference reference-amendment\\"></span>
      <h4 class=\\"heading\\"></h4>
      <div class=\\"subsection\\"><p class=\\"text-element\\">Jos eläinlääkärinammatin harjoittaja on tuomittu vankeusrangaistukseen rikoksesta, jonka hän on tehnyt ammattitoiminnassaan, ja tuomioistuimen tuomio on saanut lainvoiman ja jos rikokseen liittyvistä asianhaaroista on havaittavissa, ettei hän ole sen luottamuksen arvoinen, jota hänen tulee nauttia, Elintarviketurvallisuusvirasto voi poistaa eläinlääkärinammatin harjoittajalta määräajaksi tai, jos asianhaarat ovat erittäin raskauttavat, lopullisesti ammatinharjoittamisoikeuden.</p>
      </div>
      <div class=\\"subsection\\"><p class=\\"text-element\\">Jos valtion tai kunnan taikka kuntayhtymän virassa oleva eläinlääkärinammatin harjoittaja on rikoksen johdosta tuomittu erotettavaksi virantoimituksesta tai viralta pantavaksi, noudatetaan vastaavasti, mitä 1 momentissa säädetään.</p>
      </div>
      <div class=\\"subsection\\"><p class=\\"text-element\\">Tuomioistuimen tulee viipymättä lähettää jäljennös 1 ja 2 momentissa tarkoitettua asiaa koskevasta pöytäkirjasta ja tuomiosta Elintarviketurvallisuusvirastolle.</p>
      </div>
      <div class=\\"subsection\\"><p class=\\"text-element\\">Jo ennen kuin tuomioistuimen tuomio, jolla eläinlääkärinammatin harjoittaja on tuomittu vankeusrangaistukseen taikka viralta pantavaksi tai virantoimituksesta erotettavaksi, on saanut lainvoiman, Elintarviketurvallisuusvirasto voi kieltää eläinlääkärinammatin harjoittajaa harjoittamasta ammattia.</p>
      </div>
      </div>",
      }
    }
  />
  <ReactTooltip
    insecure={true}
    resizeHide={true}
    wrapper="div"
  />
  <hr />
</div>
`;
